From: James McCoy <jamessan@debian.org>
Date: Fri, 3 Nov 2017 19:42:16 -0400
Subject: test_search: WaitFor g:lines to be visible

On slower systems, the arbitrary wait isn't long enough.  Waiting for
the desired behavior is more reliable.

Signed-off-by: James McCoy <jamessan@debian.org>
---
 src/testdir/test_search.vim | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/src/testdir/test_search.vim b/src/testdir/test_search.vim
index 37661c3..a93ab05 100644
--- a/src/testdir/test_search.vim
+++ b/src/testdir/test_search.vim
@@ -480,13 +480,12 @@ func Test_search_cmdline8()
     return
   endif
   " Prepare buffer text
-  let lines = ['abb vim vim vi', 'vimvivim']
-  call writefile(lines, 'Xsearch.txt')
+  let g:lines = ['abb vim vim vi', 'vimvivim']
+  call writefile(g:lines, 'Xsearch.txt')
   let g:buf = term_start([GetVimProg(), '--clean', '-c', 'set noswapfile', 'Xsearch.txt'], {'term_rows': 3})
 
-  call term_wait(g:buf, 200)
-  call assert_equal(lines[0], term_getline(g:buf, 1))
-  call assert_equal(lines[1], term_getline(g:buf, 2))
+  call WaitFor('g:lines == [term_getline(g:buf, 1), term_getline(g:buf, 2)]')
+  unlet g:lines
 
   call term_sendkeys(g:buf, ":set incsearch hlsearch\<cr>")
   call term_sendkeys(g:buf, ":14vsp\<cr>")
@@ -610,9 +609,7 @@ func Test_search_cmdline_incsearch_highlight_attr()
   let g:lines = ['abb vim vim vi', 'vimvivim']
   call writefile(g:lines, 'Xsearch.txt')
   let g:buf = term_start([GetVimProg(), '--clean', '-c', 'set noswapfile', 'Xsearch.txt'], {'term_rows': 3})
-  call WaitFor('g:lines[0] == term_getline(g:buf, 1)')
-  call assert_equal(g:lines[0], term_getline(g:buf, 1))
-  call assert_equal(g:lines[1], term_getline(g:buf, 2))
+  call WaitFor('g:lines == [term_getline(g:buf, 1), term_getline(g:buf, 2)]')
   unlet g:lines
 
   " Get attr of normal(a0), incsearch(a1), hlsearch(a2) highlight
